app/models/category.rb

```diff
@@ -3,4 +3,5 @@
 class Category < ApplicationRecord
   validates :name, presence: true, uniqueness: true
   belongs_to :user, optional: true
+  has_many :events, dependent: :destroy
```

Comment:
Please ALWAYS place those things in normal order. It helps to identify everything at the first look. F.e.
```suggestion
  has_many :events, dependent: :destroy

  belongs_to :user, optional: true

  validates :name, presence: true, uniqueness: true
```